x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -554,6 +561,16 @@ public QueryBuilder semanticQuery(InferenceResults inferenceResults, float boost
                             );
                         }
 
+                        final Integer requestSize = searchExecutionContext.requestSize();
+                        if (Objects.equals(requestSize, 0)) {
+                            // Rewrite to a query that will return all docs with indexed embeddings for the queried field.
+                            // Size is often set to 0 when calculating aggregations and an unconstrained kNN query will return all docs
+                            // with indexed embeddings for the queried field, so we rewrite to a query that achieves that as efficiently as
+                            // possible.
+                            // TODO: Apply pre-filters once they are available
+                            yield new ExistsQueryBuilder(inferenceResultsFieldName);
+                        }
+
                         yield new KnnVectorQueryBuilder(inferenceResultsFieldName, inference, null, null, null);
```

Comment:
I do think another valid option is that we just pick a `k` and default to it with `max(size, DEFAULT_K)`, no matter what size is. Then, once users can set `k` themselves, they can override our default.

I do think having different defaults for semantic text vs. knn is OK.

Reply:
I agree different defaults for `semantic` vs. `knn` is ok. However, if you look at the SDH this issue came from, the customer's use case is to perform aggregations using the `semantic` query with `size: 0` (as is generally advised in such a case). Using `max(size, DEFAULT_K)` would essentially break that use case without providing a way to set the value of `k`, which is a much more complicated change and outside the scope of a bug fix.

Reply:
> Using max(size, DEFAULT_K) would essentially break that use case without providing a way to set the value of k,

No, it won't break that case. kNN's document filter for aggregation is the nearest neighbors found. 

Folks do `size: 0` all the time with `knn` queries, they can just set the `k` there to a number they care about. However, semantic query is all about simplicity and sane defaults. Consequently, having a default for `k` can work perfectly fine.

Reply:
I prefer the default size option better than switching to an exists query. The latter completely changes the profile of the query and the aggs result so we should avoid it.